src/db/migrate.rs

```diff
@@ -1,11 +1,29 @@
 //! Database migrations
 
-use db::connect_db;
 use error::Result as CratesfyiResult;
-use postgres::error::Error as PostgresError;
-use postgres::transaction::Transaction;
+use postgres::{Connection, transaction::Transaction, Error as PostgresError};
 use schemamama::{Migration, Migrator, Version};
 use schemamama_postgres::{PostgresAdapter, PostgresMigration};
+use std::rc::Rc;
+
+
+enum ApplyMode {
+    Permanent,
+    Temporary,
+}
+
+struct MigrationContext {
+    apply_mode: ApplyMode,
+}
+
+impl MigrationContext {
+    fn format_query(&self, query: &str) -> String {
+        query.replace("{create_table}", match self.apply_mode {
+            ApplyMode::Permanent => "CREATE TABLE",
+            ApplyMode::Temporary => "CREATE TEMPORARY TABLE",
+        })
```

Comment:
This seems kind of magic to me (i.e. I won't remember how it works in a week). Not sure I see a better way to reuse all these functions though ...

Reply:
Hmm, I'm out of ideas on this as well.

cc @QuietMisdreavus, do you have any better idea on how to dynamically switch from `CREATE TABLE` to `CREATE TEMPORARY TABLE` dynamically based on the `ApplyMode` in migrations?

Reply:
The only thing I can think of is to use an in-memory database connection, but I don't think there's a way to set that up from the client side... could we pass in a `Transaction` instead of a `Connection` and set [`read_only`](https://docs.rs/postgres/0.15.2/postgres/transaction/struct.Config.html#method.read_only) to true?

Reply:
@jyn514 what do you think about c7a036a? With it we won't worry about temporary tables when writing a new migration, but we can still use them behind the scenes.

Reply:
I like it! I wouldn't have thought of that, fewer changes are always nice :)